### DISCUSSION TITLE
Image upload trouble

### DISCUSSION THREAD

**Discussion by juliodlg** [neuropoly/axondeepseg#727] [Help needed]
When I try to upload an image using the "Load PNG or TIF file" button on AxonDeepSeg, the image is black. It shows that there is an uploaded image in the overlay list, and there are coordinates where the image should be, but the image is not there. 
I tried uploading both PNG or TIF image.

**Comment by mathieuboudreau** [comment]
@juliodlg thanks for opening this discussion about your problem! Could you share some additional details, maybe like which operating system you're using, and if you opened FSLeyes using a terminal, maybe a screenshot of the information in that terminal after you've tried opening the image?

Thanks!

**Reply by juliodlg** [reply]
Thank you for answering! I am using MacOS Venture 13. I opened FSLeyes using the terminal. I attached a screenshot. 
<img width="1440" alt="Screenshot 2023-03-08 at 12 11 07 PM" src="https://user-images.githubusercontent.com/127346862/223838266-cc52f789-af3e-4492-b0b7-96b90285b142.png">

**Reply by mathieuboudreau** [reply]
@Stoyan-I-A do you have any insights on of this is might be an FSLeyes issue vs plugin issue? I see that FSLeyes recently released a new major version: https://pypi.org/project/fsleyes/#history

**Reply by juliodlg** [reply]
I tried re installing the software, but I keep getting the same results. When I run the installation quick test, I get the "Integrity test passed. AxonDeepSeg is correctly installed " message.

**Reply by mathieuboudreau** [reply]
Thanks @juliodlg for checking that! The main visualization software, FSLeyes, is developed by someone else (https://github.com/pauldmccarthy/fsleyes), and we only developed a plugin for FSLeyes. So if the problem is with FSLeyes, the problem wouldn't get flagged during our tests like you checked.

**Reply by Stoyan-I-A** [reply]
@mathieuboudreau @juliodlg It looks like some update to FSLeyes made the call to colormaps behave differently. This is something we should fix on our side. I will upload a fix for this.

**Reply by Stoyan-I-A** [reply]
~~It is from this change I think~~
![image](https://user-images.githubusercontent.com/43790661/223910176-c457096c-ae8e-4c6c-aecc-4eea1e02697e.png)

Edit: actually this is not the case. Since the newest version of FSLeyes isn't compatible with our matplotlib version, we are using FSLeyes 1.5.0

**Reply by Stoyan-I-A** [reply]
Hmm strangely, on my end, images can be displayed correctly, even if the colormap is invalid. 
@juliodlg What are the dimensions of your image? I know that if an image is very large (images that are over 1 Gb) the software won't be able to display it. 
Could you also try to load an image using FSLeyes' built-in image loader (in `File -> Add from file`) and try to change its colormap?

@mathieuboudreau I have uploaded a patch (https://github.com/axondeepseg/axondeepseg/pull/728) but I don't like it, I don't know if it works on other systems and I can't reproduce the issue on my end.

**Reply by mathieuboudreau** [reply]
>@mathieuboudreau I have uploaded a patch (https://github.com/axondeepseg/axondeepseg/pull/728) but I don't like it, I don't know if it works on other systems and I can't reproduce the issue on my end.

An alternative @Stoyan-I-A could be that we fix the FSLeyes version in our venv to any version prior to the most recent release. What do you think?

**Reply by Stoyan-I-A** [reply]
> An alternative @Stoyan-I-A could be that we fix the FSLeyes version in our venv to any version prior to the most recent release. What do you think?

I don't think this will help because we are already on an older version of FSLeyes (1.5.0) because of our dependencies which are incompatible with the most recent FSLeyes release.

**Reply by juliodlg** [reply]
@Stoyan-I-A The dimension of the image i was trying to upload is 2160 × 1848. It is also 3.9 MB. I tried uploading it through FSLeyes built-in image loader, but it also just shows a black image.

**Reply by Stoyan-I-A** [reply]
>  I tried uploading it through FSLeyes built-in image loader, but it also just shows a black image.

Thank you for testing this. This information leads me to believe that the problem is on the FSLeyes' side rather than a problem with the AxonDeepSeg plugin.

There's one final test we can do to rule out the possibility of being a problem with the AxonDeepSeg plugin:
We can try to install the latest version of FSLeyes in an environment **without** AxonDeepseg and see if it can load your image. Here are the steps:

1. Open a terminal and create a new virtual environment using conda (make sure you are **NOT** in the AxonDeepSeg folder):
`conda create -n imageLoadTest python=3.8`
2. Activate the environment:
`conda activate imageLoadTest`
3. Install FSLeyes using conda-forge:
`yes | conda install -c conda-forge fsleyes`
4. Launch FSLeyes:
`fsleyes`
5. Use the built-in image loader to open your image
`File -> Add from file -> (select and open your image)`

If after theses steps the image still appears black, then I suggest opening an issue on the FSLeyes GitHub page (https://github.com/pauldmccarthy/fsleyes).

**Reply by juliodlg** [reply]
I was able to upload an image from the built-in image loader after following these steps.

**Reply by Stoyan-I-A** [reply]
Okay, I will have to check if there's a way for us to change our requirements so that we can use the latest version of FSLeyes since it seems like you won't have this problem on it.

**Reply by juliodlg** [reply]
Perfect! Thank you for your help!

**Reply by Stoyan-I-A** [reply]
@juliodlg We just updated our dependencies. I can't guarantee that this will fix your issue but it is worth trying since you were able to see your image with the latest FSLeyes version. Can you try to use the plugin now? Here are the steps:

1. Open a terminal and reach the AxonDeepSeg folder that you cloned a few days ago
`cd axondeepseg`
2. Get the latest changes
`git pull`
The rest of the steps will be similar to the regular installation
3. Create a new virtual environment 
`conda env create -f environment.yml -n ads_venv_patched`
4. Activate the environment and setup AxonDeepSeg
`conda activate ads_venv_patched`
`pip install -e .`
5. Open FSLeyes
`fsleyes`
6. The plugin should already be installed since you did it a few days ago and we didn't change the plugin file.
`Settings -> Ortho View -> ADScontrol`
7. Use the "load png or tif file" button to load your image.
